string.c

```diff
@@ -4346,11 +4346,25 @@ rb_str_byteindex_m(int argc, VALUE *argv, VALUE str)
 }
 
 #ifdef HAVE_MEMRCHR
+#define rb_memrchr memrchr
+#else
+static void*
+rb_memrchr(const char *search_str, int chr, long search_len)
+{
+    const char *ptr = search_str + search_len;
+    do {
+       if (*(--ptr) == chr) return (void *)ptr;
```

Comment:
```suggestion
        if ((unsigned char)*(--ptr) == chr) return (void *)ptr;
```

Reply:
I like this approach better, thank you for the suggestion.